core/iwasm/interpreter/wasm_interp_classic.c

```diff
@@ -3151,6 +3235,13 @@ wasm_interp_call_func_bytecode(WASMModuleInstance *module,
     FETCH_OPCODE_AND_DISPATCH ();
 #endif
 
+  call_func_from_return_call:
+    POP(cur_func->param_cell_num);
+    FREE_FRAME(exec_env, frame);
+    word_copy(frame->lp, frame_sp, cur_func->param_cell_num);
```

Comment:
This works, but logically, we had better `FREE_FRAME` after `word_copy`
```suggestion
    word_copy(frame->lp, frame_sp, cur_func->param_cell_num);
    FREE_FRAME(exec_env, frame);
```

Reply:
FREE_FRAME() first is because the tail call logic, pop the stack frame then call. More clear way may be FREE_FRAME(), outs_area to get stack_top and then word_copy() to it. But actually the outs_area operations could be saved by using current frame directly, and this is also suggested by @wenyongh in ppt. :) So less codes or clear logic?

Reply:
Yes, we should use `frame->lp` directly rather than get `outs_area`. 

My concern here is: the `frame->lp` is referenced after `FREE_FRAME(exec_env, frame)`, we know this will work just because we know `FREE_FRAME` will not modify the content in the memory, but how about we don't know? Just like dereference a freed pointer `free(ptr); a = *ptr;`. On most platforms `free` will not set `ptr` to NULL, but we never do this.

How do you think of this? @wenyongh